tests/sps30-uart-test.cpp

```diff
@@ -0,0 +1,163 @@
+#include "sensirion_test_setup.h"
+#include "sps30.h"
+
+// Measurement ranges according to datasheet
+#define SPS30_MIN_MC 0
+#define SPS30_MAX_MC 1000
+#define SPS30_MIN_NC 0
+#define SPS30_MAX_NC 3000
+#define CMD_DELAY_USEC 20000
+#define SLEEP_WAKE_UP_DELAY_USEC 5000
+#define SENSIRION_UART_TTYDEV "/dev/ttyUSB1"
+
+// Less or equal (<=) for float arithmetics with small error (1e-5)
+#define LEQ(l, u) ((l) - (u) < 1e-5)
+
+TEST_GROUP (SPS30_Test) {
+    void setup() {
+        int16_t error;
+        error = sensirion_uart_open();
+        CHECK_ZERO_TEXT(error, "sensirion_uart_open");
+    }
+
+    void teardown() {
+        int16_t error;
+        error = sps30_reset();
+        CHECK_ZERO_TEXT(error, "sps30_reset in test reset");
+        sensirion_sleep_usec(CMD_DELAY_USEC);
+        sensirion_uart_close();
+        CHECK_ZERO_TEXT(error, "sensirion_uart_close");
+    }
+};
+
+TEST (SPS30_Test, SPS30_probe) {
+    int16_t error;
+    error = sps30_probe();
+    CHECK_ZERO_TEXT(error, "sps30_probe");
+    sensirion_sleep_usec(CMD_DELAY_USEC);
+}
+
+TEST (SPS30_Test, SPS30_get_driver_version) {
+    const char* version = sps_get_driver_version();
+    CHECK_ZERO_TEXT(!version, "sps30_get_driver_version");
+    printf("sps30_get_driver_version: %s\n", version);
+}
+
+TEST (SPS30_Test, SPS30_get_serial) {
+    int16_t error;
+    char serial[SPS30_MAX_SERIAL_LEN];
+    error = sps30_get_serial(serial);
+    CHECK_ZERO_TEXT(error, "sps30_get_serial");
+    sensirion_sleep_usec(CMD_DELAY_USEC);
+    printf("SPS30 serial: %s\n", serial);
+}
+
+TEST (SPS30_Test, SPS30_sleep_and_wake_up) {
+    int16_t error;
+    error = sps30_sleep();
+    CHECK_ZERO_TEXT(error, "SPS30_sleep");
+    sensirion_sleep_usec(SLEEP_WAKE_UP_DELAY_USEC);
+    sensirion_sleep_usec(1000000);  // let sensor sleep for some time
+    error = sps30_wake_up();
+    CHECK_ZERO_TEXT(error, "SPS30_wake_up");
+    sensirion_sleep_usec(SLEEP_WAKE_UP_DELAY_USEC);
+}
+
+TEST (SPS30_Test, SPS30_fan_auto_cleaning_interval) {
+    int16_t error;
+    uint32_t get_interval;
+    uint32_t set_interval = 4 * 24 * 60 * 60;  // arbitrary
+    error = sps30_set_fan_auto_cleaning_interval(set_interval);
+    CHECK_ZERO_TEXT(error, "sps30_set_fan_auto_cleaning_interval");
+    sensirion_sleep_usec(CMD_DELAY_USEC);
+    error = sps30_get_fan_auto_cleaning_interval(&get_interval);
+    CHECK_ZERO_TEXT(error, "sps30_get_fan_auto_cleaning_interval");
+    sensirion_sleep_usec(CMD_DELAY_USEC);
+    CHECK_EQUAL_TEXT(set_interval, get_interval,
+                     "Fan auto cleaning intervals do not match");
+}
+
+TEST (SPS30_Test, SPS30_get_fan_auto_cleaning_interval_days) {
+    int16_t error;
+
```

Comment:
vspace after the decl. block would be nicer than here. I would welcome this change in every function.